libs/application/template-api-modules/src/lib/modules/templates/parental-leave/parental-leave.service.ts

```diff
@@ -99,13 +105,113 @@ export class ParentalLeaveService {
     return attachments
   }
 
+  async createPeriodsDTO(
+    application: Application,
+    nationalRegistryId: string,
+  ): Promise<Period[]> {
+    const { periods: answers } = getApplicationAnswers(application.answers)
+    const selectedChild = getSelectedChild(
+      application.answers,
+      application.externalData,
+    )
+
+    if (!selectedChild) {
+      throw new Error('Missing selected child')
+    }
+
+    const periods: Period[] = []
+    let daysAllowed = selectedChild.remainingDays
+    let extraDaysToTransformIntoPeriod = 0
+
+    for (const period of answers) {
+      const startDate = new Date(period.startDate)
+      const endDate = new Date(period.endDate)
+      const getPeriodLength = await this.parentalLeaveApi.parentalLeaveGetPeriodLength(
+        { nationalRegistryId, startDate, endDate, percentage: period.ratio },
+      )
+      const periodLength = Number(getPeriodLength?.periodLength ?? 0)
+
+      if (daysAllowed > periodLength) {
+        periods.push({
+          from: period.startDate,
+          to: period.endDate,
+          ratio: Number(period.ratio),
+          approved: false,
+          paid: false,
```

Comment:
Unrelated but I would like to get rid of these from our payloads if that makes sense (need to ask VMST).